src/operators.jl

```diff
@@ -215,13 +215,13 @@ specifies in which subsystems the corresponding operator is defined.
 function embed(basis_l::CompositeBasis, basis_r::CompositeBasis,
                operators::Dict{<:Vector{<:Integer}, T}) where T<:AbstractOperator
     @assert length(basis_l.bases) == length(basis_r.bases)
-    N = length(basis_l.bases)
+    N = length(basis_l.bases)::Int # type assertion to help type inference
     if length(operators) == 0
         return identityoperator(T, basis_l, basis_r)
     end
     indices, operator_list = zip(operators...)
     operator_list = [operator_list...;]
-    indices_flat = [indices...;]
+    indices_flat = [indices...;]::Vector{Int} # type assertion to help type inference
```

Comment:
This particular type assertion hint might be unnecessary in the future if improvements to Base.Generator happen. See https://github.com/aviatesk/JET.jl/issues/413